status-models/src/main/java/com/telecominfraproject/wlan/status/equipment/report/models/ManagedNeighbourEquipmentInfo.java

```diff
@@ -59,28 +60,22 @@ public void setRadioInfo(RadioType radioType, NeighbourRadioInfo neighbourRadioI
 	public Integer getAverageSignalStrenght(RadioType radioType)
 	{
 	   NeighbourRadioInfo info = getRadioInfo(radioType);
-	   
-	   int runningTotal = 0;
+	   Integer returnValue = null; 
 	   int numEntries = 0;
+	   int[] rssiList;
 	   
 	   if(info != null)
 	   {
+		  rssiList = new int[info.getBssIds().size()];
 	      for(NeighbourBssidInfo singleBssidInfo :  info.getBssIds())
 	      {
-	         runningTotal += singleBssidInfo.getSignal();
-	         numEntries++;
+	         rssiList[numEntries++] = singleBssidInfo.getRssi();   
 	      }
+		  if (numEntries != 0) {
+			  returnValue = (int) Math.round(DecibelUtils.getAverageDecibel(rssiList));
```

Comment:
While this works as it is, I would suggest making DecibelUtils more convenient - add a new method getAverageDecibel(Collection<Integer> decibels), and use that instead of copying lists into arrays (which is more error-prone)